packages/client/hmi-client/src/components/resources/resources-list.vue

```diff
@@ -30,25 +30,34 @@ const filteredRouteMetadata: {
 
 const assetAmount = props.resourceRoute ? 10 : 2;
 
-filteredRouteMetadata.forEach((metadata) => {
-	const route = metadata.route;
-	const { displayName, icon, projectAsset } = metadata;
-	if (projectAsset && activeProjectAssets !== null && props?.project !== null) {
-		const projId = props?.project.id;
-		const assets = activeProjectAssets[projectAsset].slice(0, assetAmount);
-		assets.forEach((asset) => {
-			resources.push({
-				route,
-				params: {
-					projectId: projId,
-					assetId: route === RouteName.DocumentRoute ? asset.xdd_uri : asset.id
-				},
-				name: displayName,
-				icon,
-				projectAsset: asset
-			});
+function updateResources() {
+	const activeProjectAssets = resourcesStore.activeProjectAssets;
+	return filteredRouteMetadata
+		.filter((metadata) => metadata.projectAsset && activeProjectAssets && props?.project)
+		.flatMap((metadata) => {
+			const { displayName, icon, projectAsset, route } = metadata;
+			const projectId = props.project?.id;
+			return (
+				activeProjectAssets![projectAsset!]?.slice(0, assetAmount)?.map((asset) => ({
+					route,
+					params: {
+						projectId,
+						assetId: route === RouteName.DocumentRoute ? asset.xdd_uri : asset.id
+					},
+					name: displayName,
+					icon,
+					projectAsset: asset
+				})) ?? []
+			);
 		});
-	}
+}
+
+resources.value = updateResources();
+
+// @ts-ignore
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+resourcesStore.$subscribe((mutation, state) => {
+	resources.value = updateResources();
 });
```

Comment:
Can you just replace `(mutation, state)` with `()` to remove all those ts and esling ignore line?
If not, then we need to look into a better solution or remove those rules.

Reply:
Done!